solutions/ceoi18-clo.mdx

```diff
@@ -0,0 +1,185 @@
+---
+id: ceoi18-clo
+source: CEOI 18
+title: Cloud Computing
+author: Kevin Sheng
+---
+
+It might be tempting at first to do a 2-dimensional DP with the number of computer purchases and the number of customer orders, but that would result in a nasty and inefficient transition.
+
+Instead, let's try grouping the two into a single array of "transactions". Each transation has three attributes- the change in the number of cores, the associated clock rate of the cores, and the change in the total profits.
+For example, the example case would have a transaction array that looks like this:
+
+<center>
+<table className="text-center">
+	<tbody>
+		<tr>
+			<td>Core change</td>
+			<td>Clock rate</td>
+			<td>Profit change</td>
+		</tr>
+		<tr>
+			<td>4</td>
+			<td>2200</td>
+			<td>-700</td>
+		</tr>
+		<tr>
+			<td>2</td>
+			<td>1800</td>
+			<td>-10</td>
+		</tr>
+		<tr>
+			<td>20</td>
+			<td>2550</td>
+			<td>-9999</td>
+		</tr>
+		<tr>
+			<td>4</td>
+			<td>2000</td>
+			<td>-750</td>
+		</tr>
+		<tr>
+			<td>-1</td>
+			<td>1500</td>
+			<td>300</td>
+		</tr>
+		<tr>
+			<td>-6</td>
+			<td>1900</td>
+			<td>1500</td>
+		</tr>
+		<tr>
+			<td>-3</td>
+			<td>2400</td>
+			<td>4550</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+Now we sort the array by clock rate in reverse, resulting in this:
+
+<center>
+<table className="text-center">
+	<tbody>
+		<tr>
+			<td>Core change</td>
+			<td>Clock rate</td>
+			<td>Profit change</td>
+		</tr>
+		<tr>
+			<td>20</td>
+			<td>2550</td>
+			<td>-9999</td>
+		</tr>
+		<tr>
+			<td>-3</td>
+			<td>2400</td>
+			<td>4550</td>
+		</tr>
+		<tr>
+			<td>4</td>
+			<td>2200</td>
+			<td>-700</td>
+		</tr>
+		<tr>
+			<td>4</td>
+			<td>2000</td>
+			<td>-750</td>
+		</tr>
+		<tr>
+			<td>-6</td>
+			<td>1900</td>
+			<td>1500</td>
+		</tr>
+		<tr>
+			<td>2</td>
+			<td>1800</td>
+			<td>-10</td>
+		</tr>
+		<tr>
+			<td>-1</td>
+			<td>1500</td>
+			<td>300</td>
+		</tr>
+	</tbody>
+</table>
+</center>
+
+Notice that in this order, as long as a customer order comes after a computer purchase, the computer will be able to satisfy the order as long as there are enough cores.
+
+This drastically simplifies the problem. Now we can simply define $\texttt{max_profits}[t][c]$ as the maximum profit we can gain from the first $t$ transactions given that we have $c$ cores left.
```

Comment:
can you be more specific what the simplified problem is

Reply:
i mean it's just knapsack at this point it's kinda obvious
i also say it later too